apisix/plugins/limit-req.lua

```diff
@@ -67,7 +67,12 @@ function _M.access(conf, ctx)
         return 500
     end
 
-    local key = (ctx.var[conf.key] or "") .. ctx.conf_type .. ctx.conf_version
+    local key
+    if conf.key == "consumer_name" then
+        key = (ctx.consumer_id or "") .. ctx.conf_type .. ctx.conf_version
```

Comment:
ok.

Reply:
I think the "consumer_id" here does not need to be checked whether it does not exist, because when the "limit-req" is bound to the "consumer", the "consumer_id" can definitely be obtained. When bound to "route", it defaults to an empty string, so that it does not affect the use of the plug-in's "key" with other values. What do you think?

Reply:
fixed

Reply:
NO. Error handling is necessary

Reply:
and test cases are also required

Reply:
ok.

Reply:
added.